### PR TITLE
Database restructuring and addition of PendingTransfer model

### PR DESCRIPTION
**Pull request by elohanlon**
Plus: Test additions and refactoring, ci.rake to wrap ci setup steps, Rubocop fixes

### CODE REVIEW

File: app/models/fixity_verification.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  belongs_to :source_object
+  belongs_to :stored_object
+
+  enum status: { pending: 0, failure: 1, success: 2 }
+
+  # Some db backends don't enforce a limit on binary field length,
+  # so the limit below is meant to ensure that we don't ever
+  # accidentally add a larger value. Make sure to update this
+  # value if the database limit ever changes.
+  validates :transfer_checksum_value, length: { maximum: 4 }
```
Comment (fcd1) [inline app/models/fixity_verification.rb:13]: Remove line 13-14 (and preceding comments)?

Comment (elohanlon) [reply app/models/fixity_verification.rb]: Good catch, thank you!  Just removed!

**Comment by elohanlon** [comment]
@fcd1 Thanks for reviewing!  Was that one comment you left the only change you thought I should make?  Or are you still reviewing?

If all good, would you mind approving this PR when you have a chance?  And then I'll go ahead and merge it.

**Review by fcd1 — APPROVED** [review]
PR approved.

**Comment by elohanlon** [comment]
Thanks!

**Pull request merged**